locked-nft/contracts/LockedNFT.cdc

```diff
@@ -0,0 +1,131 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract LockedNFT {
+
+    pub event ContractInitialized()
+    pub event NFTLocked(
+        id: UInt64,
+        to: Address?,
+        lockedAt: UInt64,
+        lockedUntil: UInt64
+    )
+    pub event NFTUnlocked(id: UInt64, from: Address?)
+
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+
+    pub var totalLockedTokens:      UInt64
+    access(self) let lockedTokens:  {UInt64: LockedData}
+
+    pub struct LockedData {
+        pub let owner: Address
+        pub let lockedAt: UInt64
+        pub let lockedUntil: UInt64
```

Comment:
Could there be any reason to also store duration here? I know it can be determined by the difference between lockedAt and lockedUntil. Maybe in the future it will be nice to have it?

Same with emitting `duration` in the event, NBA uses this to verify that the user committed to a full year to be eligible for leaderboards, etc.

Reply:
Yes. Will add that. Thanks.

Reply:
Done. Thanks.